apps/publikator/pages/repo/[owner]/[repo]/raw.js

```diff
@@ -113,61 +114,28 @@ export default withDefaultSSR(
 
     const onSave = (e) => {
       if (e) e.preventDefault()
-      const editedMdast = editMeta
-        ? parse(md)
-        : {
-            ...parse(md),
-            meta,
-          }
-
-      if (!editedMdast.meta.template) {
-        editedMdast.meta.template = meta.template
-      }
-      store.set('editorState', editedMdast)
+      store.set('editorState', safeParse(stringifiedMdast))
       goToEditor()
     }
 
-    const onEditMeta = () => {
-      if (editMeta) return
-      setEditMeta(true)
-      const contentAndMeta = stringify({
-        ...parse(md),
-        meta,
-      })
-      setMd(contentAndMeta)
-    }
-
     useEffect(() => {
       const storeKey = [repoId, commitId].join('/')
       setStore(initLocalStore(storeKey))
     }, [repoId, commitId])
 
     useEffect(() => {
       if (!store) return
-      setMeta(store.get('editorState').meta)
       const documentSchema = store.get('editorState').meta?.template || schema
       setFoldCode(documentSchema !== 'front')
-      const editorMdast = { ...store.get('editorState'), meta: {} }
-      setMd(stringify(editorMdast))
+      setStringifiedMdast(stringify(store.get('editorState')))
```

Comment:
Why are we stringifying the state here only to parse it again in onSave?

If it's to store in localStorage, I think we should only stringify/parse right there, not anywhere else in the code.

I may be missing something here 😅

Reply:
Is it because this is the raw editor?

Reply:
OK! 🫡